src/tiled/mainwindow.cpp

```diff
@@ -816,6 +817,27 @@ bool MainWindow::saveFileAs()
     return saveFile(fileName);
 }
 
+void MainWindow::saveAll()
+{
+    foreach (MapDocument *mapDoc, mDocumentManager->documents()) {
+        QString fileName(mapDoc->fileName());
+        bool successfullSave = true;
+        QString error;
+        if (fileName.isEmpty()) {
+            successfullSave = saveFileAs();
```

Comment:
You can't just call `saveFileAs`, you first need to set the `mapDoc` as the current document using `mDocumentManager->switchToDocument(mapDocument);`.